wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/AbstractHttpClientWagon.java

```diff
@@ -354,6 +366,97 @@ private static PoolingHttpClientConnectionManager createConnManager()
         return connManager;
     }
 
+    /**
+     * The type of the retry handler, default to DefaultHttpRequestRetryHandler.
+     * Values can be default, standard (StandardHttpRequestRetryHandler), or a fully qualified name class.
+     *
+     * @since 3.0.1
+     */
+    private static final String RETRY_HANDLER_CLASS =
+            System.getProperty( "maven.wagon.http.retryhandler.class", "auto" );
+
+    /**
+     * true if it's OK to retry non-idempotent requests that have been sent.
+     * Note: only used for default and standard retry handlers.
+     *
+     * @since 3.0.1
+     */
+    private static final boolean RETRY_HANDLER_REQUEST_SENT_ENABLED =
+            Boolean.getBoolean( "maven.wagon.http.retryhandler.requestSentEnabled" );
+
+    /**
+     * Number of retries for the retry handler.
+     * Note: only used for default and standard retry handlers.
+     *
+     * @since 3.0.1
+     */
+    private static final int RETRY_HANDLER_COUNT =
+            Integer.getInteger( "maven.wagon.http.retryhandler.count", 3 );
+
+    private static final String DEFAULT_NON_RETRYABLE_CLASSES =
+            InterruptedIOException.class.getName() + ","
+            + UnknownHostException.class.getName() + ","
+            + ConnectException.class.getName() + ","
+            + SSLException.class.getName();
+    /**
+     * Comma separated list of non retryable classes.
+     * Note: only used for default retry handler.
+     *
+     * @since 3.0.1
+     */
+    private static final String RETRY_HANDLER_EXCEPTIONS =
+            System.getProperty( "maven.wagon.http.retryhandler.nonRetryableClasses", DEFAULT_NON_RETRYABLE_CLASSES );
+
+    private static HttpRequestRetryHandler createRetryHandler()
+    {
+        switch ( RETRY_HANDLER_CLASS )
+        {
+            case "default":
+                if ( DEFAULT_NON_RETRYABLE_CLASSES.equals( RETRY_HANDLER_EXCEPTIONS ) )
+                {
+                    return new DefaultHttpRequestRetryHandler(
+                            RETRY_HANDLER_COUNT, RETRY_HANDLER_REQUEST_SENT_ENABLED );
+                }
+                return new DefaultHttpRequestRetryHandler(
```

Comment:
Is is really smart to provide something like this for unsafe methods by default?

Reply:
changed the default so guess it is ok now